### PR TITLE
Fix initialization of authenticator

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes WOOMOB-902

## Description

We got crash reports in the `LoginViewController.showLoginEpilogue` step of the login flow. This method can fail in two cases:
- When the `navigationController` of the view controller cannot be found.
- When the `authenticationDelegate` is nil.

This PR mitigates the issue with some enhancements:
- Updates the initialization of the authenticator to allow initializing only once [upon app start](https://github.com/woocommerce/woocommerce-ios/blob/c9c8b485b4cb3f4d80fd7e32be0d1731eeaddad3/WooCommerce/Classes/ViewRelated/AppCoordinator.swift#L69). Reasons behind this fix:
  - It's not necessary to initialize more than once.
  - The previous implementation allows the private instance to be recreated, breaking the delegate reference.
- Add messages to the `fatalError` in the login flow to better understand the issue if the above doesn't fix the crash. 

Note: I'm targeting 23.0 for this PR as the crash is likely an edge case. Let me know if you think we should target the beta build 22.9 instead.

## Testing steps

Test login using different scenarios:
- Log out of the app and try logging back in => confirm that login succeeds.
- Test login with magic link when the app is forced closed (detailed steps [here](https://github.com/woocommerce/woocommerce-ios/pull/8570)).
- Test that connecting to existing site flow still works (detailed steps [here](https://github.com/woocommerce/woocommerce-ios/pull/13766))

## Testing information

Tested the above test cases using simulator iPhone 16 iOS 18.4.

## Screenshots

No UI changes.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/JetpackSetup/JetpackSetupCoordinator.swift [hunk 1]
```diff
@@ -57,10 +57,6 @@ final class JetpackSetupCoordinator {
         self.stores = stores
         self.analytics = analytics
         self.featureFlagService = featureFlagService
-
-        /// the authenticator needs to be initialized with configs
-        /// to be used for requesting authentication link and handle login later.
-        WordPressAuthenticator.initializeWithCustomConfigs(dotcomAuthScheme: dotcomAuthScheme)
```
Comment (RafaelKayumov) [inline WooCommerce/Classes/ViewRelated/JetpackSetup/JetpackSetupCoordinator.swift:63]: Looks like the `dotcomAuthScheme` argument of `initializeWithCustomConfigs` method is no longer being passed and the `ApiCredentials.dotcomAuthScheme` default value is always used there. May be it's worth removing the method argument at all?

File: WooCommerce/Classes/ViewRelated/AppCoordinator.swift [hunk 2]
```diff
@@ -200,19 +199,13 @@ private extension AppCoordinator {
...
     }
 
-    /// Configures the WPAuthenticator and sets the authenticator UI as the window's root view.
-    func configureAndDisplayAuthenticator() {
-        configureAuthenticator()
-        displayAuthenticator()
-    }
-
     /// Configures the WPAuthenticator for usage in both logged-in and logged-out states.
     func configureAuthenticator() {
```
Comment (RafaelKayumov) [inline WooCommerce/Classes/ViewRelated/AppCoordinator.swift:210]: There is another `configureAuthenticator` call on line 292. Should it removed as well?

**Review by RafaelKayumov — APPROVED** [review]
LGTM, pre-approving. Left a couple of comments.

File: WooCommerce/Classes/ViewRelated/JetpackSetup/JetpackSetupCoordinator.swift (hunk 1 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/JetpackSetup/JetpackSetupCoordinator.swift:63]: Removed in 40282f68879d3506dab5defbdd9a03a47dd8a83e.

File: WooCommerce/Classes/ViewRelated/AppCoordinator.swift (hunk 2 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/AppCoordinator.swift:210]: Thanks for catching this! Removed in b4f38e6b36b40e94f839b61d5034f98a5024a74b.
